cmd/nodecmd/list.go

```diff
@@ -39,29 +41,41 @@ func list(_ *cobra.Command, _ []string) error {
 	if len(clustersConfig.Clusters) == 0 {
 		ux.Logger.PrintToUser("There are no clusters defined.")
 	}
-	for clusterName, clusterConf := range clustersConfig.Clusters {
+	clusterNames := maps.Keys(clustersConfig.Clusters)
+	sort.Strings(clusterNames)
+	for _, clusterName := range clusterNames {
+		clusterConf := clustersConfig.Clusters[clusterName]
 		ux.Logger.PrintToUser("Cluster %q (%s)", clusterName, clusterConf.Network.Name())
 		if err := checkCluster(clusterName); err != nil {
 			return err
 		}
-		ansibleHostIDs, err := ansible.GetAnsibleHostsFromInventory(app.GetAnsibleInventoryDirPath(clusterName))
-		if err != nil {
-			return err
-		}
-		ansibleHosts, err := ansible.GetHostMapfromAnsibleInventory(app.GetAnsibleInventoryDirPath(clusterName))
-		if err != nil {
-			return err
-		}
-		for _, ansibleHostID := range ansibleHostIDs {
-			_, cloudHostID, err := models.HostAnsibleIDToCloudID(ansibleHostID)
+		for _, cloudID := range clusterConf.GetCloudIDs() {
+			nodeConfig, err := app.LoadClusterNodeConfig(cloudID)
 			if err != nil {
 				return err
 			}
-			nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudHostID))
-			if err != nil {
-				return err
+			nodeIDStr := "----------------------------------------"
+			funcs := []string{}
+			if clusterConf.MonitoringInstance != cloudID {
+				nodeID, err := getNodeID(app.GetNodeInstanceDirPath(cloudID))
+				if err != nil {
+					return err
+				}
+				nodeIDStr = nodeID.String()
+				if clusterConf.IsAPIHost(cloudID) {
```

Comment:
nit: abstract adding `API` `Node` and `Monitor` to some function, so output is consistent everywhere